### PR TITLE
fix GC aot on 32bit target

### PR DESCRIPTION
**Pull request by xujuntwt95329**

### CODE REVIEW

File: core/iwasm/aot/aot_runtime.c [hunk 1]
```diff
@@ -3151,9 +3151,11 @@ aot_alloc_frame(WASMExecEnv *exec_env, uint32 func_index)
             }
         }
 
-        if (func_type->param_count == 0) {
+        if (func_type->param_cell_num < 2) {
             /* We reserve at least two cells for native frame */
-            frame_ref[0] = frame_ref[1] = 0;
+            for (; j < 2; j++) {
+                frame_ref[j] = 0;
+            }
```
Comment (wenyongh) [inline core/iwasm/aot/aot_runtime.c:3158]: In 32-bit, when param_cell_num = 1 and the first param is GC ref, frame_ref[0] should be 1? How about:
```C
    for (j = func_type->param_cell_num; j < 2; j++) {
         frame_ref[j] = 0;
    }
```

File: core/iwasm/compilation/aot.c [hunk 2]
```diff
@@ -466,6 +511,19 @@ aot_create_comp_data(WASMModule *module, const char *target_arch,
     }
 #endif
 
+    if (!target_arch) {
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)            \
+    || defined(BUILD_TARGET_AARCH64) || defined(BUILD_TARGET_RISCV64_LP64D) \
+    || defined(BUILD_TARGET_RISCV64_LP64)
+        is_64bit_target = true;
+#endif
```
Comment (wenyongh) [inline core/iwasm/compilation/aot.c:519]: How about:
```C
#if UINTPTR_MAX == UINT64_MAX
    is_64bit_target = true;
#endif
```

File: core/iwasm/compilation/aot_emit_variable.c [hunk 3]
```diff
@@ -83,7 +83,7 @@ aot_compile_op_set_or_tee_local(AOTCompContext *comp_ctx,
 
     if (comp_ctx->aot_frame) {
         /* Get the slot index */
-        n = comp_ctx->aot_frame->cur_wasm_func->local_offsets[local_idx];
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_emit_variable.c:86]: Could we also remove cur_wasm_func field in AOTCompFrame? Seems it is only used here.

File: core/iwasm/compilation/aot.c [hunk 4]
```diff
@@ -466,6 +511,19 @@ aot_create_comp_data(WASMModule *module, const char *target_arch,
...
+    }
+    else {
+        if (strstr(target_arch, "64") != NULL) {
```
Comment (wenyongh) [inline core/iwasm/compilation/aot.c:522]: Could we add comment about why we check it with "64"?

File: core/iwasm/aot/aot_runtime.c (hunk 1 above)
Comment (xujuntwt95329) [reply core/iwasm/aot/aot_runtime.c]: Done, thanks!

File: core/iwasm/compilation/aot.c (hunk 2 above)
Comment (xujuntwt95329) [reply core/iwasm/compilation/aot.c]: Good idea! updated.

File: core/iwasm/compilation/aot_emit_variable.c (hunk 3 above)
Comment (xujuntwt95329) [reply core/iwasm/compilation/aot_emit_variable.c:86]: Yes, removed

File: core/iwasm/compilation/aot.c (hunk 4 above)
Comment (xujuntwt95329) [reply core/iwasm/compilation/aot.c:521]: Done

File: core/iwasm/compilation/aot_llvm.h [hunk 5]
```diff
@@ -114,8 +114,6 @@ typedef struct AOTValueSlot {
 typedef struct AOTCompFrame {
     /* The current wasm module */
     WASMModule *cur_wasm_module;
-    /* The current wasm function */
-    WASMFunction *cur_wasm_func;
     /* The current wasm function index */
     uint32 cur_wasm_func_idx;
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_llvm.h:118]: Can also remove cur_wasm_func_idx, it isn't used also.

**Review by wenyongh — COMMENTED** [review]
LGTM

File: core/iwasm/compilation/aot_llvm.h (hunk 5 above)
Comment (xujuntwt95329) [reply core/iwasm/compilation/aot_llvm.h]: Removed, thanks

**Pull request merged**